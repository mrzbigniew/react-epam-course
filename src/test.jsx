import React from 'react';

import './test.scss';

const statuses = {
    hovered: 'hovered',
    normal: 'normal'
}
export class TestComponent extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            className: statuses.normal
        };
    }

    _onMouseEnter = () => {
        const state = Object.assign({}, this.state);
        state.className = statuses.hovered;
        this.setState(state);
    }

    _onMouseLeave = () => {
        const state = Object.assign({}, this.state);
        state.className = statuses.normal;
        this.setState(state);
    }

    render() {
        return (
            <div className={this.state.className}
                 onMouseEnter={this._onMouseEnter}
                 onMouseLeave={this._onMouseLeave}
            >
                {this.props.children /* eslint-disable-line */}
            </div>
        )
    }
}
