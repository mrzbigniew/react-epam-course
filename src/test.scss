.hovered {
    color: red;
}
.normal {
    color: black;
}
