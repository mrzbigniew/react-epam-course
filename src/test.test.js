import React from 'react';
import renderer from 'react-test-renderer';

import { TestComponent } from './test';

test('Test element class change when hovered', () => {
    const component = renderer.create(
        <TestComponent>
            Some text
        </TestComponent>
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
