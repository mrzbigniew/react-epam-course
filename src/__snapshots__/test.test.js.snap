// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test element class change when hovered 1`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Some text
</div>
`;

exports[`Test element class change when hovered 2`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Some text
</div>
`;

exports[`Test element class change when hovered 3`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Some text
</div>
`;
